internal/cmd/plugin/destroy/cmd.go

```diff
@@ -42,12 +46,60 @@ func NewCmd() *cobra.Command {
 			}
 
 			keepPVC, _ := cmd.Flags().GetBool("keep-pvc")
+			force, _ := cmd.Flags().GetBool("force")
+			if !force {
+				if err := ensureOtherRunningInstances(ctx, clusterName, node); err != nil {
+					return err
+				}
+			}
 			return Destroy(ctx, clusterName, node, keepPVC)
 		},
 	}
 
 	destroyCmd.Flags().BoolP("keep-pvc", "k", false,
 		"Keep the PVC but detach it from instance")
 
+	destroyCmd.Flags().BoolP("force", "f", false,
+		"Force the deletion, even if it is the last remaining instance")
+
 	return destroyCmd
 }
+
+func ensureOtherRunningInstances(ctx context.Context, clusterName string, node string) error {
+	// List all pods for the cluster
+	var podList corev1.PodList
+	if err := plugin.Client.List(ctx, &podList, client.InNamespace(plugin.Namespace), client.MatchingLabels{
+		utils.ClusterLabelName: clusterName,
+		utils.PodRoleLabelName: string(utils.PodRoleInstance),
+	}); err != nil {
+		return fmt.Errorf("error listing pods for cluster %s: %v", clusterName, err)
+	}
```

Comment:
Hmm, its not enough IMHO, because, admit you have a cluster with 2 instances.
You destroy one instance.
Then try to remove the second immediatly, here you will be ask  for confirmation.
But, If you retry few second later, when the join pod come and create the new PVC, here you will be able to delete without confirmation the last pod/pvc with the last instance of the data.

Reply:
I agree with you @cyrinux, but I also think you won't lose your data if you delete all the Pods.

What do you think of considering just the ready PVCs by checking the `cnpg.io/pvcStatus=ready` annotation?